packages/formstr-app/src/containers/CreateFormNew/providers/FormBuilder/index.tsx

```diff
@@ -75,58 +83,190 @@ export default function FormBuilderProvider({
   const [questionsList, setQuestionsList] = useState<Array<Field>>([
     generateQuestion(),
   ]);
-  const [questionIdInFocus, setQuestionIdInFocus] = useState<
-    string | undefined
-  >();
-  const [formSettings, setFormSettings] =
-    useState<IFormSettings>(InitialFormSettings);
-
+  const [questionIdInFocus, setQuestionIdInFocus] = useState<string | undefined>();
+  const [formSettings, setFormSettings] = useState<IFormSettings>(InitialFormSettings);
   const [isRightSettingsOpen, setIsRightSettingsOpen] = useState(false);
   const [isLeftMenuOpen, setIsLeftMenuOpen] = useState(false);
   const [formName, setFormName] = useState<string>(
     "This is the title of your form! Tap to edit."
   );
   const bottomElement = useRef<HTMLDivElement>(null);
-  const [relayList, setRelayList] = useState(
-    getDefaultRelays().map((relay) => {
-      return { url: relay, tempId: makeTag(6) };
-    })
-  );
-  const { pubkey: userPubkey, requestPubkey } = useProfileContext();
+  const { pubkey: userPubkey } = useProfileContext();
   const [editList, setEditList] = useState<Set<string>>(
     new Set(userPubkey ? [userPubkey] : [])
   );
   const [viewList, setViewList] = useState<Set<string>>(new Set([]));
-  const [selectedTab, setSelectedTab] = useState<string>(
-    HEADER_MENU_KEYS.BUILDER
-  );
+  const [selectedTab, setSelectedTab] = useState<string>(HEADER_MENU_KEYS.BUILDER);
   const [secretKey, setSecretKey] = useState<string | null>(null);
   const [viewKey, setViewKey] = useState<string | null | undefined>(null);
+  const navigate = useNavigate();
+
+  const [relayList, setRelayList] = useState<RelayItem[]>([]);
+  const [relayConnectionStatuses, setRelayConnectionStatuses] = useState<Map<string, RelayStatus>>(new Map());
+  const [isRelayManagerModalOpen, setIsRelayManagerModalOpen] = useState(false);
+  const [isRelayListInitialized, setIsRelayListInitialized] = useState(false);
 
+  const relayListRef = useRef(relayList);
   useEffect(() => {
-    if (userRelays.length) {
-      setRelayList(
-        userRelays.map((relay) => {
-          return { url: relay, tempId: makeTag(6) };
-        })
-      );
+    relayListRef.current = relayList;
+  }, [relayList]);
+
+  const updateRelayConnectionStatus = useCallback((relayId: string, status: RelayStatus) => {
```

Comment:
Yeah let's move it to it's own modal, this is bloating up formbuilder provider which is already very bloated